File: sdk/python/http_client/stop_profile.py
"""Stop profile using stdlib http.client."""
import os
import json
import http.client

HOST = os.getenv("MLX_LAUNCHER_HOST", "launcher.mlx.yt")
PORT = int(os.getenv("MLX_LAUNCHER_PORT", "45001"))
PROFILE_ID = os.environ["MLX_PROFILE_ID"]
path = f"/api/v1/profile/stop/p/{PROFILE_ID}"

conn = http.client.HTTPSConnection(HOST, PORT, timeout=60)
conn.request("GET", path, headers={"Accept": "application/json"})
res = conn.getresponse()
body = res.read().decode()
if res.status >= 400:
    raise RuntimeError(f"{res.status} {body}")
print(json.loads(body))

File: sdk/nodejs/stop_profile.mjs
const host = process.env.MLX_LAUNCHER_HOST ?? "launcher.mlx.yt";
const port = process.env.MLX_LAUNCHER_PORT ?? "45001";
const profileId = process.env.MLX_PROFILE_ID;

if (!profileId) throw new Error("Set MLX_PROFILE_ID");

const url = `https://${host}:${port}/api/v1/profile/stop/p/${profileId}`;
const res = await fetch(url, { headers: { Accept: "application/json" } });
if (!res.ok) throw new Error(`${res.status} ${await res.text()}`);
console.log(await res.json());
